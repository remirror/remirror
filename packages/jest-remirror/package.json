{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
