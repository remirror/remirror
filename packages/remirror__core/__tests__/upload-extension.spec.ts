import { extensionValidityTest, renderEditor } from 'jest-remirror';

import {
  ActionType,
  DecorationsExtension,
  findUploadPlaceholderPayload,
  hasUploadingFile,
  setUploadPlaceholderAction,
  uniqueId,
} from '..';

extensionValidityTest(DecorationsExtension);

test('add single placeholder', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
    dom,
  } = editor;
  const id = uniqueId('file-placeholder-');
  const { state } = add(doc(p('upload<cursor> placeholder')));
  const tr = setUploadPlaceholderAction(state.tr, {
    id,
    pos: state.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });
  view.dispatch(tr);

  expect(dom.outerHTML).toMatchSnapshot();
});

test('add multiple placeholder', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
    dom,
  } = editor;
  const id1 = uniqueId('file-placeholder-id1');
  const id2 = uniqueId('file-placeholder-id2');
  const { tr } = add(doc(p('upload<cursor> placeholder')));

  setUploadPlaceholderAction(tr, {
    id: id1,
    pos: tr.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });

  setUploadPlaceholderAction(tr, {
    id: id2,
    pos: tr.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });

  view.dispatch(tr);

  expect(dom.outerHTML).toMatchSnapshot();
});

test('remove single placeholder', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
    dom,
  } = editor;
  const id1 = uniqueId('file-placeholder-');
  const { tr } = add(doc(p('upload<cursor> placeholder')));

  setUploadPlaceholderAction(tr, {
    id: id1,
    pos: tr.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });

  view.dispatch(tr);

  const newTr = view.state.tr;
  setUploadPlaceholderAction(newTr, {
    id: id1,
    type: ActionType.REMOVE_PLACEHOLDER,
  });
  view.dispatch(newTr);

  expect(dom.outerHTML).toMatchSnapshot();
});

test('remove multiple placeholder', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
    dom,
  } = editor;
  const id1 = uniqueId('file-placeholder-');
  const id2 = uniqueId('file-placeholder-');
  const { tr } = add(doc(p('upload<cursor> placeholder')));

  setUploadPlaceholderAction(tr, {
    id: id1,
    pos: tr.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });

  setUploadPlaceholderAction(tr, {
    id: id2,
    pos: tr.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });

  view.dispatch(tr);

  const newTr = view.state.tr;

  setUploadPlaceholderAction(newTr, {
    id: id1,
    type: ActionType.REMOVE_PLACEHOLDER,
  });

  setUploadPlaceholderAction(newTr, {
    id: id2,
    type: ActionType.REMOVE_PLACEHOLDER,
  });
  view.dispatch(newTr);

  expect(dom.outerHTML).toMatchSnapshot();
});

test('hasUploadingFile function', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
  } = editor;
  const id = uniqueId('file-placeholder-');
  const { state } = add(doc(p('upload<cursor> placeholder')));
  const tr = setUploadPlaceholderAction(state.tr, {
    id,
    pos: state.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: null,
  });
  view.dispatch(tr);

  const flag = hasUploadingFile(view.state);

  expect(flag).toBeTrue();
});

test('findUploadPlaceholderPayload function', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
  } = editor;
  const id = uniqueId('file-placeholder-');
  const { state } = add(doc(p('upload<cursor> placeholder')));
  const tr = setUploadPlaceholderAction(state.tr, {
    id,
    pos: state.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: { a: 1, b: 2 },
  });
  view.dispatch(tr);

  const payload = findUploadPlaceholderPayload(view.state, id);

  expect(payload).toEqual({ a: 1, b: 2 });
});

test('update placeholder', () => {
  const editor = renderEditor<never>([]);
  const {
    nodes: { doc, p },
    add,
    view,
  } = editor;
  const id = uniqueId('file-placeholder-');
  const { state } = add(doc(p('upload<cursor> placeholder')));
  const tr = setUploadPlaceholderAction(state.tr, {
    id,
    pos: state.selection.from,
    type: ActionType.ADD_PLACEHOLDER,
    payload: { a: 1, b: 2 },
  });

  setUploadPlaceholderAction(tr, {
    id,
    type: ActionType.UPDATE_PLACEHOLDER,
    payload: { c: 3, d: 4 },
  });
  view.dispatch(tr);

  const payload = findUploadPlaceholderPayload(view.state, id);

  expect(payload).toEqual({ c: 3, d: 4 });
});
