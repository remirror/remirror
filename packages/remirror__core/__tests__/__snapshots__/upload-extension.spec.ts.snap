// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add multiple placeholder 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    upload
    <placeholder class="ProseMirror-widget">
    </placeholder>
    <placeholder class="ProseMirror-widget">
    </placeholder>
    placeholder
  </p>
</div>
`;

exports[`add single placeholder 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    upload
    <placeholder class="ProseMirror-widget">
    </placeholder>
    placeholder
  </p>
</div>
`;

exports[`remove multiple placeholder 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    upload placeholder
  </p>
</div>
`;

exports[`remove single placeholder 1`] = `
<div
  contenteditable="true"
  role="textbox"
  aria-multiline="true"
  aria-label
  translate="no"
  class="ProseMirror remirror-editor"
>
  <p>
    upload placeholder
  </p>
</div>
`;
